test/lib/utils.go

```diff
@@ -32,3 +42,27 @@ func GetEnvOrFail(t *testing.T, key string) string {
 	}
 	return value
 }
+
+// WaitForSourceAuthCheckPending polls the status of the Source from client
+// every interval until isSourceAuthCheckPending returns `true` indicating
+// it is done, or returns an error, or timeout.
+func WaitForSourceAuthCheckPending(dynamicClient dynamic.Interface, obj *resources.MetaResource) error {
+	return wait.PollImmediate(interval, timeout, func() (bool, error) {
+		return checkSourceAuthCheckPending(dynamicClient, obj)
+	})
+}
+
+func checkSourceAuthCheckPending(dynamicClient dynamic.Interface, obj *resources.MetaResource) (bool, error) {
+	psObj, err := duck.GetGenericObject(dynamicClient, obj, &apiduckv1.PubSub{})
+	return isSourceAuthCheckPending(psObj, err)
+}
+
+func isSourceAuthCheckPending(psObj runtime.Object, err error) (bool, error) {
+	if err != nil {
+		// Return error to stop the polling.
+		return false, err
+	}
```

Comment:
I would expect this to be done in the parent, rather than here.

```go
	psObj, err := duck.GetGenericObject(dynamicClient, obj, &apiduckv1.PubSub{})
	if err != nil {
		return false, err
	}
	return isSourceAuthCheckPending(psObj)
```